extensions/src/AWSSDK.Extensions.NETCore.Setup/ClientFactory.cs

```diff
@@ -219,39 +219,37 @@ private static ClientConfig CreateConfig(Type serviceInterfaceType, AWSOptions o
             }
 
             var defaultConfig = options.DefaultClientConfig;
-            if (options.IsDefaultClientConfigSet)
-            {
-                var emptyArray = new object[0];
-                var singleArray = new object[1];
+            var emptyArray = new object[0];
+            var singleArray = new object[1];
 
-                var clientConfigTypeInfo = typeof(ClientConfig).GetTypeInfo();
-                foreach (var property in clientConfigTypeInfo.DeclaredProperties)
+            var clientConfigTypeInfo = options.DefaultClientConfig.GetType();
+            var properties = clientConfigTypeInfo.GetProperties(BindingFlags.Public | BindingFlags.Instance);
```

Comment:
I changed the uses of `GetTypeInfo` to be `GetType` + `GetProperties` with binding flags in order to get inherited properties as well as ones on the declaring type.

This is also probably good to do because `GetTypeInfo` is possibly going to be deprecated as a result of this https://github.com/dotnet/runtime/issues/53217

Reply:
Since options is gauranteed to have a value I agree that it's fine to use `GetType() `here